### ISSUE TITLE
UnsupportedOperationException for immutable collection when NullValuePropertyMappingStrategy.IGNORE 

### ISSUE THREAD

**Issue by yelhouti** [mapstruct/mapstruct#3708]
### Expected behavior

When I retrieve my data from DB, sometime the collection is Immutable. ex: when using @Type(type = "json").
When I do a partial request, I expect it to keep working.
ie: Not try to clear the immutableCollection just create a new Empty one and set it.

### Actual behavior

```
        if ( entity.getTypes() != null ) {
            Set<PartnerType> set = dto.getTypes();
            if ( set != null ) {
                entity.getTypes().clear();
                entity.getTypes().addAll( set );
            }
        }
        else {
            Set<PartnerType> set = dto.getTypes();
            if ( set != null ) {
                entity.setTypes( new LinkedHashSet<PartnerType>( set ) );
            }
        }
```

### Steps to reproduce the problem

Have the target contain an ImmutableCollection ex using `Set.of()`
```
    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
    void partialUpdate(@MappingTarget Partner entity, PartnerDTO dto);
```
Try to map.

### MapStruct Version

Mapstruct 1.5.3

**Comment by filiphr** [comment]
@yelhouti have you looked into `CollectionMappingStrategy#TARGET_IMMUTABLE`?

**Comment by yelhouti** [comment]
@filiphr I must admit I didn't, I was looking for something at the level of the `@Mapping` annotation since only one of the collections is Immutable. Wouldn't that make more sense tu be able to override at that level ?

**Comment by filiphr** [comment]
>Wouldn't that make more sense tu be able to override at that level ?

You are right, I think that it does make sense to support it on `@BeanMapping` and perhaps even on `@Mapping` level as well. I'll change the title of the issue.

**Comment by yelhouti** [comment]
@filiphr I faced the same issue again today and this is still open, I guess I can give it a try if it is ok with you.

**Comment by filiphr** [comment]
Sure go ahead @yelhouti

**Comment by znight1020** [comment]
Hi! @filiphr I've just submitted a pull request(#3935) to address this. 
Would you mind taking a look when you get a chance?

**Comment by UnleashSpirit** [comment]
Hello, any news about this enhancement and the PR ?
I'm in same case, requiring a collectionMappingStrategy at @BeanMapping level :)
It will avoid a new mapper for specific case

**Comment by altaiezior** [comment]
@filiphr are there any updates available on this?
